### PR TITLE
Fix junit reporter test duplication issue for WDIO v8

### PR DESCRIPTION
**Pull request by jemishgopani**
## Proposed changes
- Added fix for the below issues for WDIO v8,
1. https://github.com/webdriverio/webdriverio/issues/10962
2. https://github.com/webdriverio/webdriverio/issues/8497

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

**Review by christian-bromann — COMMENTED** [review]
Can we have a test for this?

**Comment by jemishgopani** [comment]
unit test result of junit reporter
image

**Comment by jemishgopani** [comment]
Let me do that after that I'll ping you

**Review by christian-bromann — APPROVED** [review]
LGTM 👍

**Comment by jemishgopani** [comment]
Hi @christian-bromann 
I have changed required things if we are agree on this change then I'll add test for wdio v7 as well

File: packages/wdio-junit-reporter/tests/reporter.test.ts [hunk 1]
```diff
@@ -224,7 +234,7 @@ describe('wdio-junit-reporter', () => {
         reporter.suites = suitesLog as any
 
         // verifies the content of the report but omits format by stripping all whitespace and new lines
-        expect(reporter['_buildJunitXml'](mochaRunnerLog as any).replace(/\s/g, '').replace(/file:\/\/\/C:/g, '')).toMatchSnapshot()
+        expect(reporter['_buildJunitXml'](mochaRunnerLog as any).replace(/\s/g, '').replace(/file:\/\//g, '').replace(/C:\//g, '')).toMatchSnapshot()
     })
 
     it('generates xml output correctly when error options are set (Cucumber-style)', () => {
```
Comment (jemishgopani) [inline packages/wdio-junit-reporter/tests/reporter.test.ts:240]: @christian-bromann Please take a look at this I feel like we don't need this test.

**Comment by jemishgopani** [comment]
Can you please go through this `generates xml output correctly when the addFileAttribute option is set` I feel like we don't need that test.

**Comment by christian-bromann** [comment]
> I feel like we don't need that test.

How so?

**Comment by jemishgopani** [comment]
> > I feel like we don't need that test.
> 
> How so?

```
it('generates xml output correctly when the addFileAttribute option is set', () => {
        reporter = new WDIOJunitReporter({ stdout: true, addFileAttribute: true })
        reporter.suites = featuresLog as any

        // verifies the content of the report but omits format by stripping all whitespace and new lines
        expect(reporter['_buildJunitXml'](mochaRunnerLog as any).replace(/\s/g, '').replace(/file:\/\//g, '').replace(/C:\//g, '')).toMatchSnapshot()
    })
```

Because that test we are running on mocha-runner `mochaRunnerLog` but the specified suites `reporter.suites = featuresLog as any` are the cucumber feature.

**Comment by jemishgopani** [comment]
@christian-bromann

**Comment by jemishgopani** [comment]
Hi @christian-bromann 
Can you please help here with what can I do with that unit test case that I mentioned above?

**Comment by christian-bromann** [comment]
> but the specified suites `reporter.suites = featuresLog as any` are the cucumber feature.

Just replace `featureLog` with `suiteLog` and verify that the test still tests what it suppose to.

**Comment by jemishgopani** [comment]
@christian-bromann Fixed junit reporter unit test please review it

**Review by christian-bromann — APPROVED** [review]
LGTM 👍

**Comment by christian-bromann** [comment]
Great job, thank you @jemishgopani

**Pull request merged**